app/views/layouts/mailer.html.erb

```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+    <style>
+      /* Email styles need to be inline */
+    </style>
+  </head>
+
+  <body>
+    <%= yield %>
```

Comment:
How does `yield` work?

Reply:
Yield accepts a block of code and it runs it. From my understanding, if we pass a block into a method that has `yield` in it as below

```
def log_hello_world
  yield
end

log_hello_world { puts "hello world" }
```
it outputs `"hello world"` to the console.

So in this case, yield will accept a template and run it in its place. Am I totally off the mark here?

Reply:
Very good. 
Thats what `yield` is. Now, in the context of `yield` in the view (html.erb), how to use it?

Reply:
We can use the `<%= yield %>` to run another block of code that is declared elsewhere.

Reply:
Not clear.
Lets try this scenario:

You want to create a main layout page in which the content of the `<body> ?? </body>` be filled in by different html depending on the URL. 

How would you code that?

Reply:
?
This is just the layout. How do you make it so that `page1.html` will be loaded into the `yield` there?

Reply:
We can have a `page_controller.rb` with the following:

```
class PageController < ApplicationController
    def page1
    end
end
```

and for instance inside `/layouts/page.html.erb` we will have

```
<!DOCTYPE html>
<html>
  <head>
    <title>Page Title</title>
    <meta name="viewport" content="width=device-width,initial-scale=1">
  </head>

  <body>
    <%= yield %>
  </body>
</html>
```

which will load up the contents from `/views/page/page1.html.erb`. But instead if we want to selectively a file into the body, we could instead use `render`. We can prefix the file name with `_` to `_page1.html.erb`, and render it in the layout with `<%= render "page/page1" %>`.

Reply:
Not accurate. But, lets move on. You should get a better understanding of this later on.

`_xxx.html.erb` is what we call a `partial`. Think of a partial as a component.
Rendering a partial to a page is the same as rendering a component in the page

---

If no Q, resolve this please & hit the re-review button